src/main/java/net/minecraftforge/client/gui/config/ListConfigScreen.java

```
@@ -0,0 +1,168 @@
+        configElementList.func_230932_a_(scrollAmount); // getScrollAmount
+        field_230705_e_.add(configElementList); // children.add
+    }
+
+    public class ListConfigElementList extends ConfigElementList {
+
+        public ListConfigElementList(ConfigScreen configScreen, Minecraft mcIn) {
+            super(configScreen, mcIn);
+            for (int i = 0; i < value.size(); i++) {
+                Object item = value.get(i);
+                func_230513_b_(createElement(i, item));
+            }
+        }
+
+        protected ConfigElement createElement(int index, Object item) {
+            Interactor<?> interactor = tryCreateInteractor(index, item);
+            if (interactor == null) {
+                ITextComponent title = new StringTextComponent(index + ": " + item + (item == null ? "" : " (" + item.getClass().getSimpleName() + ")"));
+                return new ListItemConfigElement(title, index, null);
+            } else {
+                ConfigElement configElement = new ListItemConfigElement(interactor.title, index, interactor);
+                configElement.setMainWidget(interactor.control);
+                return configElement;
+            }
+        }
+
+        @Nullable
+        protected Interactor<?> tryCreateInteractor(int index, Object item) {
+            IFormattableTextComponent title = new StringTextComponent(Integer.toString(index));
+            ControlCreator creator = configScreen.getControlCreator();
+            Interactor<?> interactor = new Interactor<>(title, item);
+            interactor.addData(ControlCreator.INITIAL_VALUE_KEY, item);
+            interactor.addData(ControlCreator.DEFAULT_VALUE_KEY, item);
+            interactor.addSaver(newValue -> setAndNotify(index, newValue));
+            try {
+                creator.createAndInitialiseInteractionWidget(interactor);
+                return interactor;
+            } catch (Exception e) {
+                return null;
```

Catching raw `Exception` and silently discarding it isn't the nicest thing, why is this needed?

---

The user won’t care about whatever went wrong and they won’t want their game to crash. Catching the exception and making the element just say “please edit this in the config file” seemed clean to me. It catches the exceptions from being unable to create a widget (e.g. an unknown value gets passed in, there’s a bug in my code or a bug in a mod that uses the ControlCreator api’s code)